Tests/EventDispatcherTests.swift

```diff
@@ -0,0 +1,198 @@
+                    expect(secondCallbackWasCalled) == true
+                }
+
+                it("doesn't execute callback for another event") {
+                    eventDispatcher.on(eventName, callback: callback)
+
+                    eventDispatcher.trigger("another-event")
+
+                    expect(callbackWasCalled) == false
+                }
+
+                it("doesn't executes callback for another context object") {
+                    let anotherDispatcher = EventDispatcher()
+
+                    eventDispatcher.on(eventName, callback: callback)
+
+                    anotherDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                }
+            }
+
+            describe("once") {
+                it("executes callback function") {
+                    eventDispatcher.once(eventName, callback: callback)
+                    eventDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == true
+                }
+
+                it("doesn't execute callback function twice") {
+                    eventDispatcher.once(eventName, callback: callback)
+
+                    eventDispatcher.trigger(eventName)
+                    callbackWasCalled = false
+                    eventDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                }
+
+                it("doesn't execute callback function when it is removed") {
+                    let listenId = eventDispatcher.once(eventName, callback: callback)
+                    eventDispatcher.off(listenId)
+                    eventDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                }
+            }
+
+            describe("listenTo") {
+                it("executes callback function for an event on a given context object") {
+                    let anotherDispatcher = EventDispatcher()
+
+                    eventDispatcher.listenTo(anotherDispatcher, eventName: eventName, callback: callback)
+                    anotherDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == true
+                }
+            }
+
+            describe("listenToOnce") {
+                it("executes callback function just one time for an event on a given context object") {
+                    let anotherDispatcher = EventDispatcher()
+
+                    eventDispatcher.listenToOnce(anotherDispatcher, eventName: eventName, callback: callback)
+                    anotherDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == true
+
+                    callbackWasCalled = false
+                    anotherDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                }
+            }
+
+            describe("off") {
+                it("doesn't executes callback function if it is removed") {
```

Comment:
This ``it`` seems that is holding a context logic, do you think that is necessary to split into a ``context`` > ``it`` ?
It could be:
```swift
describe("#off") {
    context("When it is removed") {
        it("doesn't execute callback") {}
    }
}
```